routes/exchange.go

```diff
@@ -1617,41 +1618,63 @@ func (fes *APIServer) GetPostsForFollowFeedForPublicKey(bav *lib.UtxoView, start
 		return nil, errors.Wrapf(err, "GetPostsForFollowFeedForPublicKey: Problem filtering out restricted public keys: ")
 	}
 
-	minTimestampNanos := uint64(time.Now().UTC().AddDate(0, 0, -2).UnixNano()) // two days ago
-	// For each of these pub keys, get their posts, and load them into the view too
-	for _, followedPubKey := range filteredPubKeysMap {
+	var postEntriesForFollowFeed []*lib.PostEntry
+	n := 1
+	for n < 6 {
+		postEntriesForFollowFeed = nil
+		daysToSubtract := int(math.Pow(4, float64(n)))
 
-		_, dbPostAndCommentHashes, _, err := lib.DBGetAllPostsAndCommentsForPublicKeyOrderedByTimestamp(
-			bav.Handle, fes.blockchain.Snapshot(), followedPubKey, false /*fetchEntries*/, minTimestampNanos, 0, /*maxTimestampNanos*/
-		)
-		if err != nil {
-			return nil, errors.Wrapf(err, "GetPostsForFollowFeedForPublicKey: Problem fetching PostEntry's from db: ")
+		minTimestampNanos := uint64(time.Now().UTC().AddDate(0, 0, -daysToSubtract).UnixNano()) // two days ago
+		maxTimestampNanos := uint64(0)
+
+		if startAfterPostHash != nil {
+			maxTimestampNanos = bav.GetPostEntryForPostHash(startAfterPostHash).TimestampNanos
+			minTimestampNanos = uint64(time.Unix(0, int64(maxTimestampNanos)).AddDate(0, 0, -daysToSubtract).UnixNano()) // casting timestamp uint64 to int64 won't crash until 2262
 		}
 
-		// Iterate through the entries found in the db and force the view to load them.
-		// This fills in any gaps in the view so that, after this, the view should contain
-		// the union of what it had before plus what was in the db.
-		for _, dbPostOrCommentHash := range dbPostAndCommentHashes {
-			bav.GetPostEntryForPostHash(dbPostOrCommentHash)
+		if n > 4 {
+			minTimestampNanos = 0
 		}
-	}
 
-	// Iterate over the view. Put all posts authored by people you follow into an array
-	var postEntriesForFollowFeed []*lib.PostEntry
-	for _, postEntry := range bav.PostHashToPostEntry {
-		// Ignore deleted or hidden posts and any comments.
-		if postEntry.IsDeleted() || (postEntry.IsHidden && skipHidden) || len(postEntry.ParentStakeID) != 0 {
-			continue
+		// For each of these pub keys, get their posts, and load them into the view too
+		for _, followedPubKey := range filteredPubKeysMap {
+
+			_, dbPostAndCommentHashes, _, err := lib.DBGetAllPostsAndCommentsForPublicKeyOrderedByTimestamp(
```

Comment:
is this query not paginated? this seems like an easy way to run out of memory

Reply:
`minTimestampNanos` and `maxTimestampNanos` are the only options available to paginate this query. That's why I went with this approach to start with small intervals and ramp up until there are enough posts